### PR TITLE
Fixes #9347: sudo management isn't update-safe

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/9347

### CODE REVIEW

**Comment by ncharles** [comment]
need to merge https://github.com/Normation/ncf/pull/615 first

File: techniques/systemSettings/userManagement/sudoParameters/3.1/sudoParameters.st [hunk 1]
```diff
@@ -129,26 +148,21 @@ bundle agent check_sudo_parameters
...
-
-    pass3.sudo_all_lines_defined.force.sudo_all_lines_deleted::
-
-      "any" usebundle => rudder_common_report("sudoParameters", "result_repaired", "sudoersFile", "None", "None", "Some lines were deleted from the sudoers file. This implies either a manual edition or an intrusion attempt");
+        ifvarclass => "sudo_parameter_edited_section_${csection_name[${sudo_index}]}_error";
 
   commands:
-
-    (sudoconfiguration_sudoers_tmp_edit_repair|sudoconfiguration_sudoers_tmp_edit_kept).pass2::
+    sudoparameters_sudoers_tmp_file_ok::
```
Comment (peckpeck) [inline techniques/systemSettings/userManagement/sudoParameters/3.1/sudoParameters.st]: This is done even when the file has not been modified. Is it what we want ?

Comment (ncharles) [inline techniques/systemSettings/userManagement/sudoParameters/3.1/sudoParameters.st]: this is as it was - and yes, we always want to check the validity of the file

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**